apps/src/lib/levelbuilder/script-editor/UnitCard.jsx

```diff
@@ -98,14 +111,65 @@ class UnitCard extends Component {
    */
   serializeLesson = lesson => {
     let s = [];
-    let t = `stage '${escape(lesson.name)}'`;
+    let t = `lesson '${escape(lesson.name)}'`;
+    if (lesson.name) {
+      t += `, display_name: '${escape(lesson.name)}'`;
+    }
     if (lesson.lockable) {
       t += ', lockable: true';
     }
+    if (lesson.visible_after) {
+      t += ', visible_after: true';
+    }
     s.push(t);
+    if (lesson.levels) {
+      lesson.levels.forEach(level => {
+        s = s.concat(this.serializeLevel(level.ids[0], level));
+      });
+    }
+    s.push('');
     return s.join('\n');
   };
 
+  serializeLevel = (id, level, active) => {
+    const s = [];
+    const key = this.props.levelKeyList[id];
+    if (/^blockly:/.test(key)) {
+      if (level.skin) {
+        s.push(`skin '${escape(level.skin)}'`);
+      }
+      if (level.videoKey) {
+        s.push(`video_key_for_next_level '${escape(level.videoKey)}'`);
+      }
+      if (level.concepts) {
+        // concepts is a comma-separated list of single-quoted strings, so do
+        // not escape its single quotes.
+        s.push(`concepts ${level.concepts}`);
+      }
+      if (level.conceptDifficulty) {
+        s.push(`level_concept_difficulty '${escape(level.conceptDifficulty)}'`);
+      }
+    }
+    let l = `level '${escape(key)}'`;
+    if (active === false) {
+      l += ', active: false';
+    }
```

Comment:
Please remove if unused - it seems like this would be unused if we are not handling variants.